crates/toml_edit/src/array.rs

```diff
@@ -303,6 +303,19 @@ impl Array {
         }
     }
 
+    /// Retains only the elements specified by the `keep` predicate.
+    ///
+    /// In other words, remove all items for which `keep(&item)` returns `false`.
+    ///
+    /// This method operates in place, visiting each element exactly once in the
+    /// original order, and preserves the order of the retained elements.
+    pub fn retain<F>(&mut self, keep: F)
+    where
+        F: FnMut(&Item) -> bool,
+    {
+        self.values.retain(keep);
```

Comment:
This should accept `Value`, rather than `Item`.  The use of `Item` is an implementation detail and not exposed in the API more generally